### PR TITLE
fix(@spartacus/core): use method GET to get a list of cms components (GH-1033)

### PR DESCRIPTION
**Pull request by bgambocjaviniar**
closes GH-1033

### CODE REVIEW

File: projects/core/src/cms/occ/occ-cms-component.adapter.ts [hunk 1]
```diff
@@ -40,26 +41,55 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
...
+              this.componentIdsFlag,
+              currentPage,
+              pageSize,
+              sort
+            );
+
+            return this.http
+              .post<CmsComponentList>(
+                this.getComponentsEndpoint(requestParams, fields),
+                componentIds,
```
Comment (WeizhengSap) [inline projects/core/src/cms/occ/occ-cms-component.adapter.ts]: 'post' requires `ComponentIDListWsDTO`, which is `{idList: string[];}`. So, the `IdList` model you removed is for this purpose.

File: projects/core/src/cms/model/componentIds.model.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface ComponentIds {
```
Comment (WeizhengSap) [inline projects/core/src/cms/model/componentIds.model.ts]: For `get`, we don't need this model. But, for `post` we need. 'post' requires ComponentIDListWsDTO, which is {idList: string[];}. So, the IdList model you removed is for this purpose.

File: projects/core/src/cms/occ/occ-cms-component.adapter.ts [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
+import { CmsComponentAdapter } from '../connectors/component/cms-component.adapter';
 import { CMS_COMPONENT_NORMALIZER } from '../connectors/component/converters';
-import { pluck } from 'rxjs/operators';
-import { PageType } from '../../model/cms.model';
+import { ComponentIds } from '../model/componentIds.model';
 
 @Injectable()
 export class OccCmsComponentAdapter implements CmsComponentAdapter {
   protected headers = new HttpHeaders().set('Content-Type', 'application/json');
+  private componentIdsFlag;
```
Comment (WeizhengSap) [inline projects/core/src/cms/occ/occ-cms-component.adapter.ts]: better not create this class variable `componentIdsFlag`. You can add a flag in function `getComponentsRequestParams`. And `componentIdsFlag` is not a good name.

File: projects/core/src/cms/occ/occ-cms-component.adapter.ts [hunk 4]
```diff
@@ -40,26 +40,52 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
...
           headers: this.headers,
         }
       )
       .pipe(
         pluck('component'),
-        this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER)
+        this.converter.pipeableMany(CMS_COMPONENT_NORMALIZER),
+        catchError(error => {
+          if (error.status === 400) {
+            requestParams = this.getComponentsRequestParams(
```
Comment (WeizhengSap) [inline projects/core/src/cms/occ/occ-cms-component.adapter.ts]: Can we move this part into a new function? So, we have 2 `loadList` functions (with different names). So, it is consistent with backend. If the "post" endpoint is removed in future, it is easy to remove one function from SPA. And the codes look more clear. No need to modify `getComponentsRequestParams`. `requestParams['componentIds']=xxx` can just be added into the `get loadList` function.

File: projects/core/src/cms/occ/occ-cms-component.adapter.ts [hunk 5]
```diff
@@ -76,13 +122,19 @@ export class OccCmsComponentAdapter implements CmsComponentAdapter {
   }
 
   private getComponentsRequestParams(
```
Comment (WeizhengSap) [inline projects/core/src/cms/occ/occ-cms-component.adapter.ts]: The names of `getComponentsRequestParams` and `getComponentRequestParams` are very confusing. I suggest, we rename `getComponentsRequestParams` to `getPaginationParams` and `getComponentRequestParams` to `getContextParams`. So in "get" functions, you call `getPaginationParams` and `getContextParams`, and make the id list string. In "post" function, only need to call `getPaginationParams` and `getContextParams`.

**Comment by RadhepS** [comment]
There is an issue where when the GET request fails, the POST request for components is sent twice.

**Comment by RadhepS** [comment]
There is an issue where when the GET request fails, the POST request for components is sent twice.

**Pull request merged**